### PR TITLE
1) Fixes and enhancements for the top navigation bar and L/H navigation

### PR DESCRIPTION
**Pull request by justasojourner**
1. Fixed and enhanced the top navigation bar by editing relevant .pages file(s) and added extra .pages files at further locations to facilitate improvements to left side menus.
2. Added link to Rocky Linux home page to the top nav menu (can be removed of course)
2. Edited files for the Books/System Administrator's Guide to remove the need for numbering prefix for sorting.
3. Added home landing page for each section of the docs site = Guides, Books, Labs

## Author checklist (to be completed by original Author)
- [ ] Is this document a good fit for the Rocky project ?
- [ ] Is this a non-English contribution? 
- [ ] Title and Author MetaTags have been inserted into the document 
- [ ] If applicable, steps and instructions have been tested to work on a real system
- [ ] Did you perform an initial self-review to fix basic typos and grammatical correctness

## Rocky Documentation checklist  (to be completed by Rocky team) 
- [x] 1st Pass (Check that document is good fit for project and author checklist completed)
- [ ] 2nd Pass (Technical Review - check for technical correctness) 
- [ ] 3rd Pass (Basic Editorial Review)
- [ ] 4th Pass (Detailed Editorial Review and Peer Review)
- [ ] 5th Pass (Include document in TOC)
- [ ] Final pass/approval (Final Review)

### CODE REVIEW

**Comment by justasojourner** [comment]
Screenshot of one of the problems found.

PR-01

**Comment by calderds** [comment]
Hi! This is probably more of a web team decision than a docs team decision, but I like the effort to fix some bugs! I like the first and third screenshots, but I might prefer the production version of the second screenshot because it says which order you should read the guides in. Could rename those to Chapter 0, Chapter 1, ... if desired. Thanks for the contribution!

**Comment by hbjydev** [comment]
It looks good to me, putting my web team hat on

**Comment by hbjydev** [comment]
@justasojourner You need to fix your merge conflicts though :)

**Comment by justasojourner** [comment]
Actually the original author of the admin guide said my way was better, he only put the numbers in there to get it to sort the way that he wanted. He doesn’t want them. So the merge should be a replace. 

But it can’t be done until/unless my other suggestions to the .pages files are accepted **and the nav method in the top level .yml is changed.**

So it’s sort of going nowhere.

**Comment by hbjydev** [comment]
@justasojourner No no I mean your git conflicts -- they need resolving before the PR can be accepted

**Comment by justasojourner** [comment]
Guys... I think there needs to be some clarification/understanding here. @hbjydev the reason my PR was/is in 'limbo' was because the navigation part that I did to fix the L/H menu (**and the _top_ menu**) **will not work** unless the navigation code in the mkdocs.yml file...

```nav:
    - Table of Contents: index.md
    - Guides:
        - ... | guides/**/
    - Books:
        - ... | books/**/
    - Labs:
        - ... | labs/**/
    - Release Notes:
        - ... | release_notes/*.md
```
**is removed and the navigation code I put in ``docs/.pages`` is put in**. But I was asked to hold off on this PR.

However... as can be seen in the screenshots the navigation layout I had made is (objectively) better. For it to work I used the features from the Awesome Pages plugin:

https://pythonrepo.com/repo/lukasgeiter-mkdocs-awesome-pages-plugin-python-documentation

if you look through the PR you will see the .pages files (as well as the 'top level' docs/.pages nav) I made for navigation.

In short IMHO the nav section in mkdocs.yml needs to be removed (it just does not look good) and the navigation changed to the way I have done it in this PR.

--FromOZ

**Comment by hbjydev** [comment]
In that case, I'd ask that this PR be merged and a separate PR be opened against the UI repo (https://github.com/rocky-linux/docs.rockylinux.org) with the changes you need there, too.

**Comment by justasojourner** [comment]
> In that case, I'd ask that this PR be merged and a separate PR be opened against the UI repo (https://github.com/rocky-linux/docs.rockylinux.org) with the changes you need there, too.

I don't mind one way or the other - or whatever needs to be done happy to go along/support.

What needs to be discussed/agreed is - objectively - after looking at the screenshots whether people think the right hand screenshots look better than the left. And it's not just the improvement of removing the numbers for the 'System Administrator's Guide' which were only put in to get the chapters in order (but it is not necessary). If one looks at the L/H menu on the installation page 

https://docs.rockylinux.org/guides/rocky-8-installation/

It can been seen that the menu has 'Guides' twice and the folders (docs with chapters) are at the bottom not the top.

I think the right side (my PR) screenshots look much better. If there is agreement then yes, as Hayden says, there needs to be changes to the UI repo as well = basically navigation removed and navigation directives be in the docs repo.

So, over to you guys.

**Comment by hbjydev** [comment]
@wsoyinka I'm happy with this PR personally, just down to you to accept the new 'workload' of managing your own nav items :)

**Comment by justasojourner** [comment]
> @wsoyinka I'm happy with this PR personally, just down to you to accept the new 'workload' of managing your own nav items  :)

All - @hbjydev, @wsoyinka, @sspencerwire 

There has been discussion on this point, I have obviously looked at how it could be done to have the L/H nav look OK if the overriding nav code is in the mkdocs.yml file - I tried all sorts of permutations it just won't work if the nav code is in mkdocs.yml.

In terms of effort (assuming the overriding nav code is removed from mkdocs.yml) I honestly don't think it is that much:

- The 'top level' menu (in docs/.pages) is **almost never ever** going to change.
- The section nav files (for example docs/guides/.pages) are not going to change that much also. For example the .pages nav file for guides has 'static' stuff like folders that are (?) never going to change followed by `...` which means everything else. **So all the other (dynamic) items don't have to be manually listed**, the `...` option will just dynamically grow the nav menu list.

```---
 nav:
     - Guides Home: index.md
     - Installation Guide: rocky-8-installation.md
     - Apache Hardened Webserver: apache_hardened_webserver
     - Development: development
     - ...
``` 

So I don't see that it would be a significant effort.

The benefit is that the navigation looks much, much, better. And, of course, we don't have to do "01. / 02. / 03." sort of things to have navigation sorted whichever way we want.

**Comment by wsoyinka** [comment]
Hi @all:

Thank you @justasojourner for the time and effort that you've put into this !!

I'd like @sspencerwire  to chime in here. 

I think I distinctively recall a recent-ish  30 min - 1hr long virtual meeting with @justasojourner , @sspencerwire and @wsoyinka  present.   And at the meeting we thoroughly explored all the reasons to adopt these changes, as well as all the reasons not to adopt them and I think we concluded on a potential hybrid approach.  
My recollection of the conclusion from that meeting was for @justasojourner  to please help work on that hybrid approach to demo on a small section of docs.r.o.  This is not all what I am seeing here in #158 .

I think we shouldn't put any additional workload or processes or undue expectations on volunteers or folks that just want to write/contribute to Rocky documentation effort. 

@sspencerwire  thoughts ?

Finally, I am also very happy and respectful with/of the work @NeilHanlon  put in to get us to this point with the nav and site re-organization. And I am happy to iterate and build on that work to make things better.  Whenever @NeilHanlon   has a spare moment,  we should absolutely get input from Neil on #158.

**Comment by justasojourner** [comment]
> My recollection of the conclusion from that meeting was for @justasojourner to please help work on that hybrid approach to demo on a small section of docs.r.o. This is not all what I am seeing here in #158 .

Hi - I did try to get it to work, trying all sorts of variations/permutations in the directives in the .pages nav files. Nothing would work because the directives in the (system top level) mkdocs.yml file would 'override' things each time. I tried many things, read the documentation for Awesome Pages. 

I did see that the construct `- ... | guides/**/` was stated by them to be designed to do filtering.

Anyway, ultimately, as far as I was able to determine if the above construct is used in the mkdocs.yml file then that overrides everything 'below' it.

> I think we shouldn't put any additional workload or processes or undue expectations on volunteers or folks that just want to write/contribute to Rocky documentation effort.

Can understand that ideally we would not want to put extra load on people who are primarily creative, I do think though that it's not that technically onerous. It could be assigned to people who have some degree of technical skill and precision, writing a `.pages` document with this sort of content...

```
nav:
     - Guides Home: index.md
     - Installation Guide: rocky-8-installation.md
     - Apache Hardened Webserver: apache_hardened_webserver
     - Development: development
     - ...
```
is a reasonable ask for a reasonably technical person. Anyway, regardless, there is the point of the technical reality that there is really no choice if we want to have good looking navigation.

**Comment by sspencerwire** [comment]
All, Yes, I do recall the meeting and I believe we discussed trying to do the "Daemons" section of the menu and organizing that to see how that would look with @justasojourner suggestions. It looks like that has been tried, but that the mkdocs.yml file is overriding any changes being made to that section. I wondered (but have not tried) separate folders for each, but my guess is that this would be equally problematic (if not more so) on the technical end of things. I can see both sides of this issue: As @justasojourner has pointed out, reasonably technical people should be able to adjust .pages files as needed. BUT this would mean, as @wsoyinka has suggested, that any goal of having totally auto-created menu system would be out the window and it would require technical intervention. I /DO/ get the deal that if @sspencerwire or @justasojourner or for that matter any of us, were hit by a bus, this would mean that those taking over would have to understand the .pages formatting going forward. 

Overall, I think that working on the .pages file is not a particular issue. I do think, though, that to avoid confusion in contribution, that those .pages files should be modified (If we elect to go that way) by editors or managers, rather than adding that task to the contributor. Keeping it as simple as possible for the contributor should be our ultimate goal. Obviously, having the navigation pages formatted in a way that consolidates and cleans things up should also be a goal. We want the documentation to be readable and organized well. My personal fear, which I aired during our meeting, is that the L/H navigation menu is going to get very messy if we don't do something to clean and organize it.

**Comment by justasojourner** [comment]
> BUT this would mean, as @wsoyinka has suggested, that any goal of having totally auto-created menu system would be out the window and it would require technical intervention. 

True, it seems it is the case that a **totally** auto-created menu system is/seems not possible, **however** _it is not the case_ that it is an 'all or nothing' scenario. The case actually is that only the **major** menu levels would need to be manual, the other (e.g. individual pages) could be 'auto-created' in the menu. So it's not the end of the world. Again this .pages file for the menu for the Guides section of the L/H menu...

```
nav:
     - Guides Home: index.md
     - Installation Guide: rocky-8-installation.md
     - Apache Hardened Webserver: apache_hardened_webserver
     - Development: development
     - ...
```  
shows that the `...` at the end means 'everything else'. So **_not  all_** the pages have to be manually specified - make sure to put a `...` at the end. The sections (folders containing multiple docs) are at the top in the order we want and named as we want, and the other docs follow.

>if @sspencerwire or @justasojourner or for that matter any of us, were hit by a bus, this would mean that those taking over would have to understand the .pages formatting going forward.

As much as I might be flattered... if I was hit by a bus or a stray meteorite tomorrow someone else could very easily pick and maintain the .pages nav files. Truely it's not difficult.

> Overall, I think that working on the .pages file is not a particular issue. I do think, though, that to avoid confusion in contribution, that those .pages files should be modified (If we elect to go that way) by editors or managers, rather than adding that task to the contributor. 

It would need someone who knew what they were doing (but it wouldn't take long to teach them) and was responsible. Just don't let the average contributor change the files. I don't think they would look for them anyway.

> the L/H navigation menu is going to get very messy if we don't do something to clean and organize it.

Mine are nice ;-)

**Comment by justasojourner** [comment]
It looks like this is not going to progress, so it may as well be closed and the PR list kept tidy.
